Add Kyma upgrade pipeline on kind (#1831)
kyma-project/test-infra: docs/prow/kind-jobs.md

# kind Jobs

> This document describes how to execute kind jobs locally without Prow cluster.

Reviewer: ```suggestion
This document describes how to trigger kind jobs locally without a Prow cluster.
```

Author: Hm... I'm not sure if the trigger is a good word - for me, it means something asynchronous. This instruction explains how to run a job on Docker, not Kubernetes, locally and in this scenario it is synchronous.

Reviewer: ok, so maybe `run` would be the most appropriate?
```suggestion
This document describes how to run kind jobs locally without a Prow cluster.
```

## Prerequisites

- [Git](https://git-scm.com/)
- [Docker](https://www.docker.com/)
- [Debian](https://www.debian.org/) machine

## Steps

### Clone repositories

> 1. Create directory for repositories:

Reviewer: ```suggestion
1. Create a directory for the repositories you are about to clone:
```

   
   ```bash
   mkdir -p "$HOME/repositories"
   ```

> 2. Clone `kyma` repository:

Reviewer: ```suggestion
2. Clone the `kyma` repository:
```

   ```bash
   git clone https://github.com/kyma-project/kyma.git "$HOME/repositories/kyma"
   ```

> 3. Clone `test-infra` repository:

Reviewer: ```suggestion
3. Clone the `test-infra` repository:
```

   ```bash
   git clone https://github.com/kyma-project/test-infra.git "$HOME/repositories/test-infra"
   ```

> ### Start job

Reviewer: ```suggestion
### Trigger a kind job
```

Reviewer: as you did not like trigger let's go with `run`:
```suggestion
### Run a kind job
```

> 1. Create `artifacts` directory:

Reviewer: ```suggestion
1. Create the `artifacts` directory:
```

   ```bash
   mkdir -p "$HOME/artifacts"
   ```

> 2. Create `docker-graph` directory:

Reviewer: ```suggestion
2. Create the `docker-graph` directory:
```

   ```bash
   mkdir -p "$HOME/docker-graph"
   ```

> 3. Run Docker container with configuration based on ProwJob definition, for example configuration for `pre-master-kyma-kind-upgrade`:

Reviewer: ```suggestion
3. Run a Docker container with configuration based on a ProwJob definition. For example, use the configuration for `pre-master-kyma-kind-upgrade`:
```

### Run a kind job

1. Create the `artifacts` directory:

   ```bash
   mkdir -p "$HOME/artifacts"
   ```

2. Create the `docker-graph` directory:

   ```bash
   mkdir -p "$HOME/docker-graph"
   ```

3. Run a Docker container with configuration based on a ProwJob definition. For example, use the configuration for `pre-master-kyma-kind-upgrade`:

```bash
docker run \
    --rm \
    --privileged \
    --volume "$HOME/repositories:/home/prow/go/src/github.com/kyma-project" \
    --volume "$HOME/artifacts:/artifacts" \
    --volume "/sys/fs/cgroup:/sys/fs/cgroup" \
    --volume "/lib/modules:/lib/modules" \
    --volume "$HOME/docker-graph:/docker-graph" \
    --env ARTIFACTS="/artifacts" \
    --env GOPATH="/home/prow/go" \
    eu.gcr.io/kyma-project/test-infra/buildpack-golang-toolbox:v20191011-51ed45a \
    /home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/kind-upgrade-kyma.sh \
    --ensure-kubectl \
    --ensure-helm \
    --update-hosts \
    --delete-cluster \
    --tune-inotify \
    --start-docker \
    --kubernetes-version \
    v1.14.6 \
    --kyma-sources \
    /home/prow/go/src/github.com/kyma-project/kyma \
    --kyma-installation-timeout \
    30m
```

> All logs and artifacts generated during job run are stored in `artifacts` directory created in step 1.

Reviewer: ```suggestion
All logs and artifacts generated during a job run are stored in the `artifacts` directory created in step 1 of the [**Run a kind job**](#Run a kind job) section.
```
